src/helpers/optionalStacking/common.ts

```diff
@@ -16,3 +18,35 @@ export function getViewByTwoAttributes(
     }
     return [viewBy] as VisualizationObject.IVisualizationAttribute[];
 }
+
+/**
+ * Show a measure as a percentage
+ * one measure
+ *      => computeRatio = true (ignore stackBy)
+ *      => computeRatio = false or not set => evaluate stackMeasuresToPercent
+ * multiple measures
+ *      => ignore computeRatio and use stackMeasuresToPercent
+ *
+ * stackMeasures is applied only when there are [2 measures and more]
+ * stackMeasuresToPercent is applied only when there are [1 measure + 1 stackBy] or [2 measures and up]
+ */
+export function getSanitizedStackingConfig(
+    afm: AFM.IAfm,
+    chartConfig: IChartConfig,
+    hasStackByAttribute: boolean,
+): IMeasuresStackConfig {
+    if (get(afm, ["measures", "length"]) === 1) {
+        const stackingConfig: IMeasuresStackConfig = {};
+        const isComputeRatio = get(afm, ["measures", "0", "definition", "measure", "computeRatio"], false);
+        if (isComputeRatio && chartConfig.stackMeasures) {
+            stackingConfig.stackMeasures = false;
+        }
+        if (isComputeRatio || !hasStackByAttribute) {
+            if (chartConfig.stackMeasuresToPercent) {
+                stackingConfig.stackMeasuresToPercent = false;
+            }
+        }
+        return stackingConfig;
+    }
+    return {};
```

Comment:
I find this a little bit strange. Shouldn't this return original config if it returns sanitized config? Based on the contract defined by the method name, I would expect this method to always return the passed config but in case it needs some sanitization, it will make changes.

Possibly structure it as follows:

```
    const stackingConfig: IMeasuresStackConfig = { ...config };
    if (get(afm, ["measures", "length"]) === 1) {    
        const isComputeRatio = get(afm, ["measures", "0", "definition", "measure", "computeRatio"], false);
        if (isComputeRatio && chartConfig.stackMeasures) {
            stackingConfig.stackMeasures = false;
        }
        if (isComputeRatio || !hasStackByAttribute) {
            if (chartConfig.stackMeasuresToPercent) {
                stackingConfig.stackMeasuresToPercent = false;
            }
        }
    }
    return stackingConfig;
```

What do you think?

Reply:
Sorry for the bad method name.
I only want to sanitize the stacking config (`stackMeasures` and `stackMeasuresToPercent`).
And I will override the whole config [here](https://github.com/gooddata/gooddata-react-components/pull/910/files#diff-400bbc09cb815ed39403371beb675951R203)

I changed the name from `getSanitizedStackingConfig` to `getSanitizedStackMeasuresConfiguration`

Reply:
@xMort, Could you suggest a better name please?

Reply:
I cannot think of one with the current functionality.
If you would return the config cloned and overridden if necessary as I suggested, you could keep the name and change the linked piece of code just to: 

````
return getSanitizedStackingConfig(afm, config, hasStackByAttribute);
````

(ChartTransformation.tsx:207)

Reply:
updated as your suggestion. Thanks :)